app/views/account/_login.html.erb

```diff
@@ -41,31 +41,33 @@ See COPYRIGHT and LICENSE files for more details.
         <div class="form--field-extra-actions">
           <% if Setting.autologin? %>
             <label class="form--label-with-check-box">
-              <%= styled_check_box_tag 'autologin', 1, false, id: 'autologin-pulldown' %> <%= t(:label_stay_logged_in) %></label>
+              <%= styled_check_box_tag 'autologin', 1, false, id: 'autologin-pulldown', tabindex: 3 %> <%= t(:label_stay_logged_in) %></label>
```

Comment:
Is there a reason this has tabindex: 3? tabindex should normally be 0 everywhere otherwise it interferes in the normal tab order of the page.

Reply:
I found it annoying to do: enter login, tab, enter password, tab, hit space to check the "Stay logged in" checbox... wait... no the submit button was focused. I'm logged in now. Ok, I'll check the box with the mouse next time.

And even if noticing that submit is focused, hitting tab multiple times hardly gets to the "Stay logged in" checbox, or to the reset password / sign up links.

login, password, and submit all had `tabindex: 1`, and other had nothing. I attributed some tabindex to have the order I wanted:
1. login
2. password
3. stay logged in
4. sign in button
5. lost password link
6. register link

I didn't know that could interfere with normal tab order and should be 0. Is there a better way to achieve the same?

Reply:
I think every `tabindex` should be 0, and then if you want to autofocus something you add the `autofocus` attribute. `tabindex` other than -1 or 0 is [problematic](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex).

> Warning: Avoid using tabindex values greater than 0. Doing so makes it difficult for people who rely on assistive technology to navigate and operate page content. Instead, write the document with the elements in a logical sequence.

You can achieve the correct tab order by writing the HTML in that order. As long as a specific input is before another one in HTML, they will be focused in that order no matter where they are rendered. `display: flex` with specific orderings (maybe even explicit `order: $num` rules) has always worked for me.

Reply:
Thanks.
I do not want to loose to much time on it right now. I may try to fix it another time.
I'll revert the tab index to what it previously was, and just fix the autofocus things.